src/load-app.ts

```diff
@@ -43,3 +43,34 @@ export async function loadApp(appPath: string = "."): Promise<App> {
     fileCache.finishedLoading();
   }
 }
+
+/**
+ * Checks to make sure that an app has enough required and distinguishing properties to determine its type.
+ */
+function isCarrierApp(definition: AppDefinition): boolean {
```

Comment:
```suggestion
function isCarrierApp(definition: AppDefinition): definition is CarrierAppDefinition {
```

Use a type guard here instead of just returning a boolean.  This causes TypeScript to treat the parameter as a `CarrierAppDefinition` type if the function returns true, so you don't have to cast stuff